Source/FLHook/HkFuncTools.cpp

```diff
@@ -216,12 +216,11 @@ HK_ERROR HkResolveShortCut(const wstring &wscShortcut, uint &_iClientID)
 
 uint HkGetClientIDByShip(uint iShip)
 {
-	for (uint i = 0; i <= MAX_CLIENT_ID; i++)
+	const CShip* cobj = reinterpret_cast<CShip*>(CObject::Find(iShip, CObject::CSHIP_OBJECT));
```

Comment:
For a single player ingame, results in miliseconds for a million repeats

Search for a nonexistent/ non-player ship:
timeCOBjectFind: 9
standardSearch: 150

Search for an existing player ship. Do note that the old search scales linearly with half of amount of players logged in, so it would take ~60 players online to catch up.
timeCOBjectFind: 62
standardSearch: 2